### PR TITLE
G-UNI and Uniswap v3 pricing

### PR DESCRIPTION
**Pull request by devinaconley**
This PR adds support for G-UNI tokens and expands pricing estimation to use Uniswap v3 pools
- detect and label G-UNI liquidity tokens
- pricing method for G-UNI tokens
- implement token pricing via uniswap v3 pool

Development subgraph with this update running here on polygon deployment:
https://thegraph.com/hosted-service/subgraph/devinaconley/gysr-dev?query=get%20pools

### CODE REVIEW

File: src/pricing/uniswap.ts [hunk 1]
```diff
@@ -123,26 +126,74 @@ export function getTokenPrice(address: Address): BigDecimal {
...
+        continue;
       }
+
+      // compute price
+      let pool = UniswapPoolV3.bind(poolAddress);
+      let slot0 = pool.slot0();
+      let sqrtPriceX96 = slot0.value0;
+      let price = (sqrtPriceX96.times(sqrtPriceX96).toBigDecimal()).div(
+        BigInt.fromI32(2).pow(96 * 2).toBigDecimal() // effective bit shift >> (96*2)
+      );
```
Comment (IainM22) [inline src/pricing/uniswap.ts:182]: Are there docs or something I could look at that explain how on earth this calculation leads to a price?

File: src/pricing/uniswap.ts [hunk 2]
```diff
@@ -1,12 +1,13 @@
 // pricing for uniswap traded tokens and uniswap LP tokens
 
-import { Address, BigInt, BigDecimal, log } from '@graphprotocol/graph-ts'
+import { Address, BigInt, BigDecimal, log, ethereum, dataSource } from '@graphprotocol/graph-ts'
```
Comment (IainM22) [inline src/pricing/uniswap.ts]: It doesn't look like either of these are being used

**Review by IainM22 — APPROVED** [review]
Very nice

File: src/pricing/uniswap.ts (hunk 1 above)
Comment (devinaconley) [reply src/pricing/uniswap.ts:182]: haha there are
https://docs.uniswap.org/sdk/guides/fetching-prices#understanding-sqrtprice

File: src/pricing/uniswap.ts (hunk 2 above)
Comment (devinaconley) [reply src/pricing/uniswap.ts]: fixed :+1:

**Comment by devinaconley** [comment]
Thanks @IainM22 !

**Pull request merged**